### PR TITLE
Upd oauth urls

### PR DESCRIPTION
**Pull request by gabemontero**
Fixes https://github.com/openshift/jenkins-openshift-login-plugin/issues/46

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1671633

### CODE REVIEW

**Comment by gabemontero** [comment]
/assign @bparees 

@enj FYI

File: src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java [hunk 1]
```diff
@@ -393,6 +393,7 @@ boolean populateDefaults() throws IOException, GeneralSecurityException {
                 // the hostname or ip/port that is only available within the
                 // cluster
                 this.defaultedRedirectURL = provider.issuer;
+                this.testProviderEndpoints(credential, transport);
```
Comment (bparees) [inline src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java]: this doesn't check the return code so what's it doing?

Comment (gabemontero) [reply src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java]: it is a diagnostic,  given what @enj mentioned about the oauth stuff moving around, coming up and down ... I can add a comment

**Comment by bparees** [comment]
nits, otherwise lgtm

File: src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java [hunk 2]
```diff
@@ -535,6 +536,55 @@ public void initialize(HttpRequest request)
                 OpenShiftProviderInfo.class);
         return info;
     }
+    
+    private boolean testProviderEndpoints(
```
[the remark this answers is not in the document]
Comment (gabemontero) [reply src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java]: Yep `useProviderOAuthEndpoint` is a better name ... I'll change

**Pull request merged**